metrics-operator/controllers/common/providers/dummy/dummy.go

```diff
@@ -0,0 +1,104 @@
+package dummy
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/go-logr/logr"
+	metricsapi "github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1beta1"
+)
+
+type KeptnDummyProvider struct {
+	Log        logr.Logger
+	HttpClient http.Client
+}
+
+func (d *KeptnDummyProvider) FetchAnalysisValue(ctx context.Context, query string, analysis metricsapi.Analysis, provider *metricsapi.KeptnMetricsProvider) (string, error) {
+	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
```

Comment:
instead of this complex code the 3 functions could return a string made out of a  printf ("dummy provider EvaluateQuery was called with query %s", query )  this way in the integration test we can assert that the metrics using the provider has that value as it's status

Reply:
For the analysis you could add data regarding the analisis itself to the string ex. timeframe and args

Reply:
Alright